### DISCUSSION TITLE
Best pattern for sending data from client to server

### DISCUSSION THREAD

**Discussion by justtonytone** [enisdenjo/graphql-ws#109] [Q&A] [answered]
We're using this library with apollo, so maybe this question makes more sense in the apollo forums.

If we have a websocket connection open already and we want to send some data back to the server, does it make sense to make a graphql mutation call that would travel through the websocket connection? Or should we instead make it a subscription that immediately closes on the server side?

**Comment by n1ru4l** [comment, marked as answer]
That sounds like you would want to do a mutation. Subscriptions are more meant for subscribing to events instead of updating stuff. That's what mutations are for, as you are probably **mutate**ing state.

Websockets are just the transport of this protocol, so you can do all available operation types (subscription, mutations, and queries) via that transport.

**Reply by justtonytone** [reply]
Thanks Laurin, that makes sense. Looks like we need to update the link being sent to apollo to optionally send non-subscription calls.
